src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -13,7 +13,34 @@ internal abstract class TagsList : ITags
 
         public List<KeyValuePair<string, double>> Metrics => Volatile.Read(ref _metrics);
 
-        public List<KeyValuePair<string, string>> Tags => Volatile.Read(ref _tags);
+        public List<KeyValuePair<string, string>> GetAll()
+        {
+            var customTags = GetCustomTags();
+            var additionalTags = GetAdditionalTags();
+            var allTags = new List<KeyValuePair<string, string>>(
+                customTags?.Count ?? 0 +
+                additionalTags?.Length ?? 0);
+
+            if (customTags != null)
+            {
+                allTags.AddRange(customTags);
```

Comment:
**race condition:**
missing `lock` on `customTags`
Notice `SetTag()` and `ToString()` methods

Reply:
Fixed